micrometer-core/src/main/java/io/micrometer/core/instrument/composite/AbstractCompositeMeter.java

```diff
@@ -30,7 +29,9 @@ abstract class AbstractCompositeMeter<T extends Meter> extends AbstractMeter imp
 
     private final AtomicBoolean childrenGuard = new AtomicBoolean();
 
-    private Map<MeterRegistry, T> children = Collections.emptyMap();
+    // IdentityHashMap instead of Collections.emptyMap() keeps implementation type
+    // consistent, which may help JIT optimization.
+    private Map<MeterRegistry, T> children = new IdentityHashMap<>(0);
```

Comment:
There are some behavioral differences between this and `Collections.emptyMap()`:
1. `Collections.emptyMap()` is unmodifiable but I guess we shouldn't use `Collections.unmodifiableMap(new IdentityHashMap<>(0))` because of the same reason this change abandons `Collections.emptyMap()`.
2. `Collections.emptyMap()` is singleton, this creates a new instance every time. Either we should use `private static final Map<MeterRegistry, T> EMPTY_CHILDREN = new IdentityHashMap<>(0);` or rework how we handle the children map.